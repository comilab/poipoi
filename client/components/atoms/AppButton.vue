<template lang="pug">
component.inline-flex.items-center.justify-center.whitespace-no-wrap.transition-colors.duration-300.disabled_opacity-75.disabled_pointer-events-none.focus_outline-none(
  :is="component",
  :to="to",
  :class="innerButtonClass",
  v-bind="innerButtonAttrs",
  v-on="$listeners",
  data-testid="button"
)
  div(
    v-if="icon !== null",
    v-bind="innerIconAttrs"
  )
    font-awesome-icon.fa-fw(
      v-if="innerIcon !== 'r18'",
      :icon="innerIcon",
      :pulse="pulse",
      :class="innerIconClass",
      data-testid="icon"
    )
    .inline-block.text-xs.font-bold.rounded.tracking-tighter.leading-tight.px-px(
      v-else-if="innerIcon === 'r18'",
      :class="innerIconClass",
      data-testid="icon"
    ) R18
  div(
    :class="innerTextClass",
    data-testid="text"
  )
    slot
</template>

<script lang="ts">
import { defineComponent, computed } from 'nuxt-composition-api'
import { PropType } from '@vue/composition-api'
import { IconDefinition } from '@fortawesome/fontawesome-common-types'
import { faSpinner } from '@fortawesome/free-solid-svg-icons'

import useClassProp, { prop as classProp } from '~/composables/use-class-prop'

export default defineComponent({
  props: {
    type: {
      type: String,
      required: false,
      default: 'button'
    },
    to: {
      type: [String, Object],
      required: false,
      default: null
    },
    href: {
      type: String,
      required: false,
      default: null
    },
    color: {
      type: String,
      required: false,
      default: null
    },
    size: {
      type: String,
      required: false,
      default: 'md'
    },
    icon: {
      type: [String, Object] as PropType<string|IconDefinition>,
      required: false,
      default: null
    },
    indeterminate: {
      type: Boolean,
      required: false,
      default: false
    },
    breakpoint: {
      type: String,
      required: false,
      default: null
    },
    pill: {
      type: Boolean,
      required: false,
      default: false
    },
    square: {
      type: Boolean,
      required: false,
      default: false
    },
    buttonClass: {
      ...classProp
    },
    iconClass: {
      ...classProp
    },
    textClass: {
      ...classProp
    }
  },
  setup (props, { attrs, slots }) {
    const { merge } = useClassProp()

    const component = computed(() => {
      if (props.type === 'badge') {
        return 'div'
      } else if (props.to) {
        return 'nuxt-link'
      } else if (props.href) {
        return 'a'
      }
      return 'button'
    })

    const innerButtonAttrs = computed(() => {
      const innerAttrs = { ...attrs }
      if (component.value === 'button') {
        innerAttrs.type = props.type
      } else if (component.value === 'a') {
        innerAttrs.href = props.href
      }
      return innerAttrs
    })

    const innerIcon = computed(() => props.indeterminate ? faSpinner : props.icon)

    const pulse = computed(() => innerIcon.value === faSpinner)

    const iconOnly = computed(() => props.icon && !slots.default)

    const innerIconAttrs = computed(() => {
      const attrs: Record<string, any> = {}
      if (pulse.value) {
        attrs.role = 'status'
      }
      return attrs
    })

    const innerButtonClass = computed(() => {
      const classes = []

      if (props.type === 'badge') {
        classes.push('pointer-events-none')
      }

      if (props.color === 'blue') {
        classes.push('bg-blue-600', 'text-white', 'hover_bg-blue-700')
      } else if (props.color === 'red') {
        classes.push('bg-red-700', 'text-white', 'hover_bg-red-800')
      } else if (props.color === 'white') {
        classes.push('bg-white', 'text-gray-600', 'hover_bg-gray-300', 'hover_text-gray-800')
      } else if (props.color === 'yellow') {
        classes.push('bg-yellow-500', 'text-gray-800', 'hover_bg-yellow-600')
      } else if (props.color === 'gray') {
        classes.push('bg-gray-300', 'text-gray-800', 'hover_bg-gray-400')
      }

      if (props.size === 'sm') {
        classes.push('text-sm', 'px-2', 'py-1')
      } else if (props.size === 'md') {
        classes.push('px-3', 'py-2')
      } else if (props.size === 'lg') {
        classes.push('text-xl', 'px-4', 'py-2')
      }

      if (iconOnly.value || props.square || props.breakpoint) {
        if (props.size === 'sm') {
          classes.push('square-sm')
        } else if (props.size === 'md') {
          classes.push('square-md')
        }
      }

      if (props.breakpoint === 'md') {
        classes.push('breakpoint-md')
      } else if (props.breakpoint === 'lg') {
        classes.push('breakpoint-lg')
      }

      if (props.pill) {
        classes.push('rounded-full')
      } else {
        classes.push('rounded')
      }

      if (props.icon === 'r18') {
        classes.push('has-r18-icon')
      }

      return merge(classes, props.buttonClass)
    })

    const innerIconClass = computed(() => {
      const classes = []

      if (!iconOnly.value) {
        if (props.breakpoint === 'md') {
          classes.push('md_mr-2')
        } else if (props.breakpoint === 'lg') {
          classes.push('lg_mr-2')
        } else {
          classes.push('mr-2')
        }
      }

      if (innerIcon.value === 'r18') {
        if (props.color === 'red') {
          classes.push('text-red-700', 'bg-white', 'hover_text-red-800')
        }
      }

      return merge(classes, props.iconClass)
    })

    const innerTextClass = computed(() => {
      const classes = []

      if (props.breakpoint === 'md') {
        classes.push('hidden', 'md_block')
      } else if (props.breakpoint === 'lg') {
        classes.push('hidden', 'lg_block')
      }

      return merge(classes, props.textClass)
    })

    return {
      component,
      innerButtonAttrs,
      innerIcon,
      pulse,
      iconOnly,
      innerIconAttrs,
      innerButtonClass,
      innerIconClass,
      innerTextClass
    }
  }
})
</script>

<style lang="postcss" scoped>
.square-sm {
  width: 29px;
  height: 29px;
}

.square-md {
  width: 37px;
  height: 37px;
}

@screen md {
  .breakpoint-md {
    @apply w-auto;
  }
}

@screen lg {
  .breakpoint-lg {
    @apply w-auto;
  }
}
</style>
