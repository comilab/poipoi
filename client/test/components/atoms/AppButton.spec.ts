import { render, RenderOptions, within } from '@testing-library/vue'

import AppButton from '~/components/atoms/AppButton.vue'

describe('components/atoms/AppButton', () => {
  let options: RenderOptions<AppButton>
  let props: any

  beforeEach(() => {
    props = {}

    options = {
      slots: {
        default: 'button-text'
      },
      stubs: ['FontAwesomeIcon', 'NuxtLink'],
      props
    }
  })

  test('renders correctly', () => {
    const { container, getByText } = render(AppButton, options)

    expect(container).toMatchSnapshot()
    getByText('button-text')
  })

  describe('component', () => {
    test('props.type = badge なら div', () => {
      props.type = 'badge'
      const { container, getByTestId } = render(AppButton, options)

      expect(container).toMatchSnapshot()
      expect(getByTestId('button')).toBeInstanceOf(HTMLDivElement)
    })

    test('props.to があれば nuxt-link', () => {
      props.to = '/to'
      const { container } = render(AppButton, options)

      expect(container).toMatchSnapshot()
      expect(container.querySelector(':scope > NuxtLink-stub')).toBeInTheDocument()
    })

    test('props.href があれば a', () => {
      props.href = '/href'
      const { container, getByTestId } = render(AppButton, options)

      expect(container).toMatchSnapshot()
      expect(getByTestId('button')).toBeInstanceOf(HTMLAnchorElement)
    })

    test('それ以外なら button', () => {
      const { container, getByTestId } = render(AppButton, options)

      expect(container).toMatchSnapshot()
      expect(getByTestId('button')).toBeInstanceOf(HTMLButtonElement)
    })
  })

  describe('type', () => {
    test('コンポーネントがbuttonならtype属性としてセット', () => {
      props.type = 'submit'
      const { container, getByTestId } = render(AppButton, options)

      expect(container).toMatchSnapshot()

      expect(getByTestId('button')).toBeInstanceOf(HTMLButtonElement)
      expect(getByTestId('button')).toHaveAttribute('type', 'submit')
    })

    test('コンポーネントがbuttonでなければtype属性はセットしない', () => {
      props.type = 'badge'
      const { container, getByTestId } = render(AppButton, options)

      expect(container).toMatchSnapshot()

      expect(getByTestId('button')).not.toBeInstanceOf(HTMLButtonElement)
      expect(getByTestId('button')).not.toHaveAttribute('type')
    })

    test('デフォルトはbutton', () => {
      const { container, getByTestId } = render(AppButton, options)

      expect(container).toMatchSnapshot()
      expect(getByTestId('button')).toHaveAttribute('type', 'button')
    })

    test('badgeならpointer-events-noneクラスをセット', () => {
      props.type = 'badge'
      const { container, getByTestId } = render(AppButton, options)

      expect(container).toMatchSnapshot()
      expect(getByTestId('button')).toHaveClass('pointer-events-none')
    })
  })

  describe('to', () => {
    test('NuxtLinkに渡す', () => {
      props.to = '/to'
      const { container } = render(AppButton, options)

      expect(container).toMatchSnapshot()

      const button = container.querySelector(':scope > NuxtLink-stub')
      expect(button).toBeInTheDocument()
      expect(button).toHaveAttribute('to', '/to')
    })
  })

  describe('href', () => {
    test('a要素の属性にセット', () => {
      props.href = '/href'
      const { container, getByTestId } = render(AppButton, options)

      expect(container).toMatchSnapshot()

      expect(getByTestId('button')).toBeInstanceOf(HTMLAnchorElement)
      expect(getByTestId('button')).toHaveAttribute('href', '/href')
    })
  })

  describe('color', () => {
    test('blue', () => {
      props.color = 'blue'
      const { container } = render(AppButton, options)

      expect(container).toMatchSnapshot()
    })

    test('red', () => {
      props.color = 'red'
      const { container } = render(AppButton, options)

      expect(container).toMatchSnapshot()
    })

    test('white', () => {
      props.color = 'white'
      const { container } = render(AppButton, options)

      expect(container).toMatchSnapshot()
    })

    test('yellow', () => {
      props.color = 'yellow'
      const { container } = render(AppButton, options)

      expect(container).toMatchSnapshot()
    })

    test('gray', () => {
      props.color = 'gray'
      const { container } = render(AppButton, options)

      expect(container).toMatchSnapshot()
    })
  })

  describe('size', () => {
    test('sm', () => {
      props.size = 'sm'
      const { container } = render(AppButton, options)

      expect(container).toMatchSnapshot()
    })

    test('md', () => {
      props.size = 'md'
      const { container } = render(AppButton, options)

      expect(container).toMatchSnapshot()
    })

    test('lg', () => {
      props.size = 'lg'
      const { container } = render(AppButton, options)

      expect(container).toMatchSnapshot()
    })

    test('sm + iconOnly', () => {
      props.size = 'sm'
      props.icon = 'icon'
      options.slots = {}
      const { container, getByTestId } = render(AppButton, options)

      expect(container).toMatchSnapshot()
      expect(getByTestId('button')).toHaveClass('square-sm')
    })

    test('md + iconOnly', () => {
      props.size = 'md'
      props.icon = 'icon'
      options.slots = {}
      const { container, getByTestId } = render(AppButton, options)

      expect(container).toMatchSnapshot()
      expect(getByTestId('button')).toHaveClass('square-md')
    })
  })

  describe('icon', () => {
    test('FontAwesomeIconに渡す', () => {
      props.icon = 'icon'
      const { container } = render(AppButton, options)

      expect(container).toMatchSnapshot()

      const button = container.querySelector(':scope > FontAwesomeIcon-stub')
      expect(button).toBeInTheDocument()
      expect(button).toHaveAttribute('icon', 'icon')
    })

    test('r18の場合はFontAwesomeを使わない', () => {
      props.icon = 'r18'
      const { container, getByTestId } = render(AppButton, options)

      expect(container).toMatchSnapshot()
      expect(getByTestId('icon')).toBeInstanceOf(HTMLDivElement)
      within(getByTestId('icon')).getByText('R18')
    })
  })

  describe('indeterminate', () => {
    beforeEach(() => {
      props.indeterminate = true
      props.icon = 'icon'
    })

    test('iconが強制的にfaSpinnerになる', () => {
      const { container, getByTestId } = render(AppButton, options)

      expect(container).toMatchSnapshot()
      expect(getByTestId('icon')).not.toHaveAttribute('icon', 'icon')
    })

    test('FontAwesomeIconにpulseを渡す', () => {
      const { container, getByTestId } = render(AppButton, options)

      expect(container).toMatchSnapshot()
      expect(getByTestId('icon')).toHaveAttribute('pulse')
    })
  })

  describe('breakpoint', () => {
    beforeEach(() => {
      props.icon = 'icon'
    })

    test('md', () => {
      props.breakpoint = 'md'
      const { container, getByTestId } = render(AppButton, options)

      expect(container).toMatchSnapshot()
      expect(getByTestId('button')).toHaveClass('breakpoint-md')
    })

    test('lg', () => {
      props.breakpoint = 'lg'
      const { container, getByTestId } = render(AppButton, options)

      expect(container).toMatchSnapshot()
      expect(getByTestId('button')).toHaveClass('breakpoint-lg')
    })
  })

  describe('pill', () => {
    beforeEach(() => {
      props.pill = true
    })

    test('true', () => {
      const { container } = render(AppButton, options)

      expect(container).toMatchSnapshot()
    })
  })

  describe('square', () => {
    beforeEach(() => {
      props.square = true
    })

    test('true + sm', () => {
      props.size = 'sm'
      const { container, getByTestId } = render(AppButton, options)

      expect(container).toMatchSnapshot()
      expect(getByTestId('button')).toHaveClass('square-sm')
    })

    test('true + md', () => {
      props.size = 'md'
      const { container, getByTestId } = render(AppButton, options)

      expect(container).toMatchSnapshot()
      expect(getByTestId('button')).toHaveClass('square-md')
    })
  })

  describe('buttonClass', () => {
    beforeEach(() => {
      props.buttonClass = 'button-class'
    })

    test('ボタンにクラスがセットされる', () => {
      const { container, getByTestId } = render(AppButton, options)

      expect(container).toMatchSnapshot()
      expect(getByTestId('button')).toHaveClass('button-class')
    })
  })

  describe('iconClass', () => {
    beforeEach(() => {
      props.iconClass = 'icon-class'
      props.icon = 'icon'
    })

    test('アイコンにクラスがセットされる', () => {
      const { container, getByTestId } = render(AppButton, options)

      expect(container).toMatchSnapshot()
      expect(getByTestId('icon')).toHaveClass('icon-class')
    })
  })

  describe('textClass', () => {
    beforeEach(() => {
      props.textClass = 'text-class'
    })

    test('テキストにクラスがセットされる', () => {
      const { container, getByTestId } = render(AppButton, options)

      expect(container).toMatchSnapshot()
      expect(getByTestId('text')).toHaveClass('text-class')
    })
  })

  describe('attrs', () => {
    beforeEach(() => {
      options.attrs = { foo: 'bar' }
    })

    test('ボタンに属性がセットされる', () => {
      const { container, getByTestId } = render(AppButton, options)

      expect(container).toMatchSnapshot()
      expect(getByTestId('button')).toHaveAttribute('foo', 'bar')
    })
  })
})
