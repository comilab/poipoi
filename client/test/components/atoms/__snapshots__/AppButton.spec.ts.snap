// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components/atoms/AppButton attrs ボタンに属性がセットされる 1`] = `
<div>
  <button
    class="inline-flex items-center justify-center whitespace-no-wrap transition-colors duration-300 disabled_opacity-75 disabled_pointer-events-none focus_outline-none px-3 py-2 rounded"
    data-testid="button"
    foo="bar"
    type="button"
  >
    <!---->
    <div
      class=""
      data-testid="text"
    >
      button-text
    </div>
  </button>
</div>
`;

exports[`components/atoms/AppButton breakpoint lg 1`] = `
<div>
  <button
    class="inline-flex items-center justify-center whitespace-no-wrap transition-colors duration-300 disabled_opacity-75 disabled_pointer-events-none focus_outline-none px-3 py-2 square-md breakpoint-lg rounded"
    data-testid="button"
    type="button"
  >
    <div>
      <fontawesomeicon-stub
        class="fa-fw lg_mr-2"
        data-testid="icon"
        icon="icon"
      />
    </div>
    <div
      class="hidden lg_block"
      data-testid="text"
    >
      button-text
    </div>
  </button>
</div>
`;

exports[`components/atoms/AppButton breakpoint md 1`] = `
<div>
  <button
    class="inline-flex items-center justify-center whitespace-no-wrap transition-colors duration-300 disabled_opacity-75 disabled_pointer-events-none focus_outline-none px-3 py-2 square-md breakpoint-md rounded"
    data-testid="button"
    type="button"
  >
    <div>
      <fontawesomeicon-stub
        class="fa-fw md_mr-2"
        data-testid="icon"
        icon="icon"
      />
    </div>
    <div
      class="hidden md_block"
      data-testid="text"
    >
      button-text
    </div>
  </button>
</div>
`;

exports[`components/atoms/AppButton buttonClass ボタンにクラスがセットされる 1`] = `
<div>
  <button
    class="inline-flex items-center justify-center whitespace-no-wrap transition-colors duration-300 disabled_opacity-75 disabled_pointer-events-none focus_outline-none px-3 py-2 rounded button-class"
    data-testid="button"
    type="button"
  >
    <!---->
    <div
      class=""
      data-testid="text"
    >
      button-text
    </div>
  </button>
</div>
`;

exports[`components/atoms/AppButton color blue 1`] = `
<div>
  <button
    class="inline-flex items-center justify-center whitespace-no-wrap transition-colors duration-300 disabled_opacity-75 disabled_pointer-events-none focus_outline-none bg-blue-600 text-white hover_bg-blue-700 px-3 py-2 rounded"
    data-testid="button"
    type="button"
  >
    <!---->
    <div
      class=""
      data-testid="text"
    >
      button-text
    </div>
  </button>
</div>
`;

exports[`components/atoms/AppButton color gray 1`] = `
<div>
  <button
    class="inline-flex items-center justify-center whitespace-no-wrap transition-colors duration-300 disabled_opacity-75 disabled_pointer-events-none focus_outline-none bg-gray-300 text-gray-800 hover_bg-gray-400 px-3 py-2 rounded"
    data-testid="button"
    type="button"
  >
    <!---->
    <div
      class=""
      data-testid="text"
    >
      button-text
    </div>
  </button>
</div>
`;

exports[`components/atoms/AppButton color red 1`] = `
<div>
  <button
    class="inline-flex items-center justify-center whitespace-no-wrap transition-colors duration-300 disabled_opacity-75 disabled_pointer-events-none focus_outline-none bg-red-700 text-white hover_bg-red-800 px-3 py-2 rounded"
    data-testid="button"
    type="button"
  >
    <!---->
    <div
      class=""
      data-testid="text"
    >
      button-text
    </div>
  </button>
</div>
`;

exports[`components/atoms/AppButton color white 1`] = `
<div>
  <button
    class="inline-flex items-center justify-center whitespace-no-wrap transition-colors duration-300 disabled_opacity-75 disabled_pointer-events-none focus_outline-none bg-white text-gray-600 hover_bg-gray-300 hover_text-gray-800 px-3 py-2 rounded"
    data-testid="button"
    type="button"
  >
    <!---->
    <div
      class=""
      data-testid="text"
    >
      button-text
    </div>
  </button>
</div>
`;

exports[`components/atoms/AppButton color yellow 1`] = `
<div>
  <button
    class="inline-flex items-center justify-center whitespace-no-wrap transition-colors duration-300 disabled_opacity-75 disabled_pointer-events-none focus_outline-none bg-yellow-500 text-gray-800 hover_bg-yellow-600 px-3 py-2 rounded"
    data-testid="button"
    type="button"
  >
    <!---->
    <div
      class=""
      data-testid="text"
    >
      button-text
    </div>
  </button>
</div>
`;

exports[`components/atoms/AppButton component props.href があれば a 1`] = `
<div>
  <a
    class="inline-flex items-center justify-center whitespace-no-wrap transition-colors duration-300 disabled_opacity-75 disabled_pointer-events-none focus_outline-none px-3 py-2 rounded"
    data-testid="button"
    href="/href"
  >
    <!---->
    <div
      class=""
      data-testid="text"
    >
      button-text
    </div>
  </a>
</div>
`;

exports[`components/atoms/AppButton component props.to があれば nuxt-link 1`] = `
<div>
  <nuxtlink-stub
    class="inline-flex items-center justify-center whitespace-no-wrap transition-colors duration-300 disabled_opacity-75 disabled_pointer-events-none focus_outline-none px-3 py-2 rounded"
    data-testid="button"
    to="/to"
  >
    <!---->
    <div
      class=""
      data-testid="text"
    >
      button-text
    </div>
  </nuxtlink-stub>
</div>
`;

exports[`components/atoms/AppButton component props.type = badge なら div 1`] = `
<div>
  <div
    class="inline-flex items-center justify-center whitespace-no-wrap transition-colors duration-300 disabled_opacity-75 disabled_pointer-events-none focus_outline-none pointer-events-none px-3 py-2 rounded"
    data-testid="button"
  >
    <!---->
    <div
      class=""
      data-testid="text"
    >
      button-text
    </div>
  </div>
</div>
`;

exports[`components/atoms/AppButton component それ以外なら button 1`] = `
<div>
  <button
    class="inline-flex items-center justify-center whitespace-no-wrap transition-colors duration-300 disabled_opacity-75 disabled_pointer-events-none focus_outline-none px-3 py-2 rounded"
    data-testid="button"
    type="button"
  >
    <!---->
    <div
      class=""
      data-testid="text"
    >
      button-text
    </div>
  </button>
</div>
`;

exports[`components/atoms/AppButton href a要素の属性にセット 1`] = `
<div>
  <a
    class="inline-flex items-center justify-center whitespace-no-wrap transition-colors duration-300 disabled_opacity-75 disabled_pointer-events-none focus_outline-none px-3 py-2 rounded"
    data-testid="button"
    href="/href"
  >
    <!---->
    <div
      class=""
      data-testid="text"
    >
      button-text
    </div>
  </a>
</div>
`;

exports[`components/atoms/AppButton icon FontAwesomeIconに渡す 1`] = `
<div>
  <button
    class="inline-flex items-center justify-center whitespace-no-wrap transition-colors duration-300 disabled_opacity-75 disabled_pointer-events-none focus_outline-none px-3 py-2 rounded"
    data-testid="button"
    type="button"
  >
    <div>
      <fontawesomeicon-stub
        class="fa-fw mr-2"
        data-testid="icon"
        icon="icon"
      />
    </div>
    <div
      class=""
      data-testid="text"
    >
      button-text
    </div>
  </button>
</div>
`;

exports[`components/atoms/AppButton icon r18の場合はFontAwesomeを使わない 1`] = `
<div>
  <button
    class="inline-flex items-center justify-center whitespace-no-wrap transition-colors duration-300 disabled_opacity-75 disabled_pointer-events-none focus_outline-none px-3 py-2 rounded has-r18-icon"
    data-testid="button"
    type="button"
  >
    <div>
      <div
        class="inline-block text-xs font-bold rounded tracking-tighter leading-tight px-px mr-2"
        data-testid="icon"
      >
        R18
      </div>
    </div>
    <div
      class=""
      data-testid="text"
    >
      button-text
    </div>
  </button>
</div>
`;

exports[`components/atoms/AppButton iconClass アイコンにクラスがセットされる 1`] = `
<div>
  <button
    class="inline-flex items-center justify-center whitespace-no-wrap transition-colors duration-300 disabled_opacity-75 disabled_pointer-events-none focus_outline-none px-3 py-2 rounded"
    data-testid="button"
    type="button"
  >
    <div>
      <fontawesomeicon-stub
        class="fa-fw mr-2 icon-class"
        data-testid="icon"
        icon="icon"
      />
    </div>
    <div
      class=""
      data-testid="text"
    >
      button-text
    </div>
  </button>
</div>
`;

exports[`components/atoms/AppButton indeterminate FontAwesomeIconにpulseを渡す 1`] = `
<div>
  <button
    class="inline-flex items-center justify-center whitespace-no-wrap transition-colors duration-300 disabled_opacity-75 disabled_pointer-events-none focus_outline-none px-3 py-2 rounded"
    data-testid="button"
    type="button"
  >
    <div
      role="status"
    >
      <fontawesomeicon-stub
        class="fa-fw mr-2"
        data-testid="icon"
        icon="[object Object]"
        pulse="true"
      />
    </div>
    <div
      class=""
      data-testid="text"
    >
      button-text
    </div>
  </button>
</div>
`;

exports[`components/atoms/AppButton indeterminate iconが強制的にfaSpinnerになる 1`] = `
<div>
  <button
    class="inline-flex items-center justify-center whitespace-no-wrap transition-colors duration-300 disabled_opacity-75 disabled_pointer-events-none focus_outline-none px-3 py-2 rounded"
    data-testid="button"
    type="button"
  >
    <div
      role="status"
    >
      <fontawesomeicon-stub
        class="fa-fw mr-2"
        data-testid="icon"
        icon="[object Object]"
        pulse="true"
      />
    </div>
    <div
      class=""
      data-testid="text"
    >
      button-text
    </div>
  </button>
</div>
`;

exports[`components/atoms/AppButton pill true 1`] = `
<div>
  <button
    class="inline-flex items-center justify-center whitespace-no-wrap transition-colors duration-300 disabled_opacity-75 disabled_pointer-events-none focus_outline-none px-3 py-2 rounded-full"
    data-testid="button"
    type="button"
  >
    <!---->
    <div
      class=""
      data-testid="text"
    >
      button-text
    </div>
  </button>
</div>
`;

exports[`components/atoms/AppButton renders correctly 1`] = `
<div>
  <button
    class="inline-flex items-center justify-center whitespace-no-wrap transition-colors duration-300 disabled_opacity-75 disabled_pointer-events-none focus_outline-none px-3 py-2 rounded"
    data-testid="button"
    type="button"
  >
    <!---->
    <div
      class=""
      data-testid="text"
    >
      button-text
    </div>
  </button>
</div>
`;

exports[`components/atoms/AppButton size lg 1`] = `
<div>
  <button
    class="inline-flex items-center justify-center whitespace-no-wrap transition-colors duration-300 disabled_opacity-75 disabled_pointer-events-none focus_outline-none text-xl px-4 py-2 rounded"
    data-testid="button"
    type="button"
  >
    <!---->
    <div
      class=""
      data-testid="text"
    >
      button-text
    </div>
  </button>
</div>
`;

exports[`components/atoms/AppButton size md + iconOnly 1`] = `
<div>
  <button
    class="inline-flex items-center justify-center whitespace-no-wrap transition-colors duration-300 disabled_opacity-75 disabled_pointer-events-none focus_outline-none px-3 py-2 square-md rounded"
    data-testid="button"
    type="button"
  >
    <div>
      <fontawesomeicon-stub
        class="fa-fw"
        data-testid="icon"
        icon="icon"
      />
    </div>
    <div
      class=""
      data-testid="text"
    />
  </button>
</div>
`;

exports[`components/atoms/AppButton size md 1`] = `
<div>
  <button
    class="inline-flex items-center justify-center whitespace-no-wrap transition-colors duration-300 disabled_opacity-75 disabled_pointer-events-none focus_outline-none px-3 py-2 rounded"
    data-testid="button"
    type="button"
  >
    <!---->
    <div
      class=""
      data-testid="text"
    >
      button-text
    </div>
  </button>
</div>
`;

exports[`components/atoms/AppButton size sm + iconOnly 1`] = `
<div>
  <button
    class="inline-flex items-center justify-center whitespace-no-wrap transition-colors duration-300 disabled_opacity-75 disabled_pointer-events-none focus_outline-none text-sm px-2 py-1 square-sm rounded"
    data-testid="button"
    type="button"
  >
    <div>
      <fontawesomeicon-stub
        class="fa-fw"
        data-testid="icon"
        icon="icon"
      />
    </div>
    <div
      class=""
      data-testid="text"
    />
  </button>
</div>
`;

exports[`components/atoms/AppButton size sm 1`] = `
<div>
  <button
    class="inline-flex items-center justify-center whitespace-no-wrap transition-colors duration-300 disabled_opacity-75 disabled_pointer-events-none focus_outline-none text-sm px-2 py-1 rounded"
    data-testid="button"
    type="button"
  >
    <!---->
    <div
      class=""
      data-testid="text"
    >
      button-text
    </div>
  </button>
</div>
`;

exports[`components/atoms/AppButton square true + md 1`] = `
<div>
  <button
    class="inline-flex items-center justify-center whitespace-no-wrap transition-colors duration-300 disabled_opacity-75 disabled_pointer-events-none focus_outline-none px-3 py-2 square-md rounded"
    data-testid="button"
    type="button"
  >
    <!---->
    <div
      class=""
      data-testid="text"
    >
      button-text
    </div>
  </button>
</div>
`;

exports[`components/atoms/AppButton square true + sm 1`] = `
<div>
  <button
    class="inline-flex items-center justify-center whitespace-no-wrap transition-colors duration-300 disabled_opacity-75 disabled_pointer-events-none focus_outline-none text-sm px-2 py-1 square-sm rounded"
    data-testid="button"
    type="button"
  >
    <!---->
    <div
      class=""
      data-testid="text"
    >
      button-text
    </div>
  </button>
</div>
`;

exports[`components/atoms/AppButton textClass テキストにクラスがセットされる 1`] = `
<div>
  <button
    class="inline-flex items-center justify-center whitespace-no-wrap transition-colors duration-300 disabled_opacity-75 disabled_pointer-events-none focus_outline-none px-3 py-2 rounded"
    data-testid="button"
    type="button"
  >
    <!---->
    <div
      class="text-class"
      data-testid="text"
    >
      button-text
    </div>
  </button>
</div>
`;

exports[`components/atoms/AppButton to NuxtLinkに渡す 1`] = `
<div>
  <nuxtlink-stub
    class="inline-flex items-center justify-center whitespace-no-wrap transition-colors duration-300 disabled_opacity-75 disabled_pointer-events-none focus_outline-none px-3 py-2 rounded"
    data-testid="button"
    to="/to"
  >
    <!---->
    <div
      class=""
      data-testid="text"
    >
      button-text
    </div>
  </nuxtlink-stub>
</div>
`;

exports[`components/atoms/AppButton type badgeならpointer-events-noneクラスをセット 1`] = `
<div>
  <div
    class="inline-flex items-center justify-center whitespace-no-wrap transition-colors duration-300 disabled_opacity-75 disabled_pointer-events-none focus_outline-none pointer-events-none px-3 py-2 rounded"
    data-testid="button"
  >
    <!---->
    <div
      class=""
      data-testid="text"
    >
      button-text
    </div>
  </div>
</div>
`;

exports[`components/atoms/AppButton type コンポーネントがbuttonでなければtype属性はセットしない 1`] = `
<div>
  <div
    class="inline-flex items-center justify-center whitespace-no-wrap transition-colors duration-300 disabled_opacity-75 disabled_pointer-events-none focus_outline-none pointer-events-none px-3 py-2 rounded"
    data-testid="button"
  >
    <!---->
    <div
      class=""
      data-testid="text"
    >
      button-text
    </div>
  </div>
</div>
`;

exports[`components/atoms/AppButton type コンポーネントがbuttonならtype属性としてセット 1`] = `
<div>
  <button
    class="inline-flex items-center justify-center whitespace-no-wrap transition-colors duration-300 disabled_opacity-75 disabled_pointer-events-none focus_outline-none px-3 py-2 rounded"
    data-testid="button"
    type="submit"
  >
    <!---->
    <div
      class=""
      data-testid="text"
    >
      button-text
    </div>
  </button>
</div>
`;

exports[`components/atoms/AppButton type デフォルトはbutton 1`] = `
<div>
  <button
    class="inline-flex items-center justify-center whitespace-no-wrap transition-colors duration-300 disabled_opacity-75 disabled_pointer-events-none focus_outline-none px-3 py-2 rounded"
    data-testid="button"
    type="button"
  >
    <!---->
    <div
      class=""
      data-testid="text"
    >
      button-text
    </div>
  </button>
</div>
`;
